### PR TITLE
[20.05] Fix history export include hidden handling

### PR DESCRIPTION
**Pull request by mvdbeek**
Fix https://github.com/galaxyproject/galaxy/issues/13386 and https://github.com/galaxyproject/galaxy/issues/13391.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: test/unit/test_model_store.py [hunk 1]
```diff
@@ -32,6 +32,30 @@ def test_import_export_history_failed_job():
...
+def test_import_export_history_hidden_false_with_hidden_dataset():
+    app = _mock_app()
+
+    u, h, d1, d2, j = _setup_simple_cat_job(app)
+    d2.visible = False
+    app.model.session.flush()
+
+    imported_history = _import_export_history(app, h, export_files="copy", include_hidden=False)
+    assert d2.dataset.get_size() > 0
+    assert imported_history.datasets[-1].get_size() == 0
```
Comment (nsoranzo) [inline test/unit/test_model_store.py]: I think it's important that we also check that the non-hidden datasets are exported/imported.

```suggestion
    assert d1.dataset.get_size() == imported_history.datasets[0].get_size()
    assert imported_history.datasets[1].get_size() == 0
```

This works locally for me, but if `get_size()` could return slightly different numbers after export/import, the first line can simply be `assert imported_history.datasets[0].get_size() > 0`

File: test/unit/test_model_store.py [hunk 2]
```diff
@@ -32,6 +32,30 @@ def test_import_export_history_failed_job():
...
+def test_import_export_history_hidden_true_with_hidden_dataset():
+    app = _mock_app()
+
+    u, h, d1, d2, j = _setup_simple_cat_job(app)
+    d2.visible = False
+    app.model.session.flush()
+
+    imported_history = _import_export_history(app, h, export_files="copy", include_hidden=True)
+    assert d2.dataset.get_size() > 0
+    assert imported_history.datasets[-1].get_size() > 0
```
Comment (nsoranzo) [inline test/unit/test_model_store.py]: As above:

```suggestion
    assert d1.dataset.get_size() == imported_history.datasets[0].get_size()
    assert d2.dataset.get_size() == imported_history.datasets[1].get_size()
```

File: test/unit/test_model_store.py (hunk 1 above)
Comment (mvdbeek) [reply test/unit/test_model_store.py]: No, changing the size wouldn't be expected, it's not going through the upload that could do newline conversions or similar data modifying things.

Comment (nsoranzo) [reply test/unit/test_model_store.py]: I was more worried about the possible difference between real file size and space used on disk.

**Review by nsoranzo — APPROVED** [review]

**Review by dannon — APPROVED** [review]

**Pull request merged**